server/src/test/java/org/opensearch/action/support/replication/ReplicationModeAwareProxyTests.java

```diff
@@ -0,0 +1,130 @@
+        ShardRouting targetRouting = TestShardRouting.newShardRouting(
+            new ShardId(new Index("test_index", "_na_"), 0),
+            "dummy-node",
+            null,
+            true,
+            ShardRoutingState.STARTED,
+            AllocationId.newInitializing("abc")
+        );
+        ShardRouting primaryRouting = TestShardRouting.newShardRouting(
+            new ShardId(new Index("test_index", "_na_"), 0),
+            "dummy-node",
+            null,
+            true,
+            ShardRoutingState.STARTED,
+            AllocationId.newInitializing("abc")
+        );
+        final ReplicationModeAwareProxy replicationModeAwareProxy = new ReplicationModeAwareProxy(
+            ReplicationMode.NO_REPLICATION,
+            DiscoveryNodes.builder().add(IndexShardTestUtils.getFakeRemoteEnabledNode("dummy-node")).build(),
+            mock(TransportReplicationAction.ReplicasProxy.class),
+            mock(TransportReplicationAction.ReplicasProxy.class)
+        );
+        assertEquals(ReplicationMode.NO_REPLICATION, replicationModeAwareProxy.determineReplicationMode(targetRouting, primaryRouting));
+    }
+
+    public void testDetermineReplicationModeTargetRoutingRelocatingPrimary() {
+        AllocationId primaryId = AllocationId.newRelocation(AllocationId.newInitializing());
+        AllocationId relocationTargetId = AllocationId.newTargetRelocation(primaryId);
+        ShardRouting targetRouting = TestShardRouting.newShardRouting(
+            new ShardId(new Index("test_index", "_na_"), 0),
+            "dummy-node-2",
+            null,
+            true,
+            ShardRoutingState.INITIALIZING,
+            relocationTargetId
+        );
+        ShardRouting primaryRouting = TestShardRouting.newShardRouting(
+            new ShardId(new Index("test_index", "_na_"), 0),
+            "dummy-node",
+            "dummy-node-2",
+            true,
+            ShardRoutingState.RELOCATING,
+            primaryId
+        );
+        final ReplicationModeAwareProxy replicationModeAwareProxy = new ReplicationModeAwareProxy(
+            ReplicationMode.NO_REPLICATION,
+            DiscoveryNodes.builder()
+                .add(IndexShardTestUtils.getFakeRemoteEnabledNode(targetRouting.currentNodeId()))
+                .add(IndexShardTestUtils.getFakeRemoteEnabledNode(primaryRouting.currentNodeId()))
+                .build(),
+            mock(TransportReplicationAction.ReplicasProxy.class),
+            mock(TransportReplicationAction.ReplicasProxy.class)
+        );
+        assertEquals(ReplicationMode.FULL_REPLICATION, replicationModeAwareProxy.determineReplicationMode(targetRouting, primaryRouting));
+    }
+
+    public void testDetermineReplicationModeTargetRoutingDocrepShard() {
+        ShardRouting primaryRouting = TestShardRouting.newShardRouting(
+            new ShardId(new Index("test_index", "_na_"), 0),
+            "dummy-node",
+            false,
```

Comment:
this should be true and the targetRouting should be false.

Reply:
Done in latest rev